performance/objcinterop/src/main/kotlin-native/org/jetbrains/objCinteropBenchmarks/complexNumbers.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.complexNumbers
+import kotlinx.cinterop.*
+import platform.posix.*
+import kotlin.math.sqrt
+import kotlin.math.PI
+import kotlin.math.cos
+import kotlin.math.sin
+import kotlin.random.Random
+import platform.Foundation.*
+import platform.darwin.*
+
+actual typealias ComplexNumber = Complex
+
+actual class ComplexNumbersBenchmark actual constructor() {
+    val complexNumbersSequence = generateNumbersSequence()
+
+    fun randomNumber() = Random.nextDouble(0.0, benchmarkSize.toDouble())
+
+    actual fun generateNumbersSequence(): List<Complex> {
+        val result = mutableListOf<Complex>()
+        for (i in 1..benchmarkSize) {
+            result.add(Complex(randomNumber(), randomNumber()))
+        }
+        return result
+    }
+
+    actual fun sumComplex() {
+        complexNumbersSequence.map { it.add(it) }.reduce { acc, it -> acc?.add(it) }
```

Comment:
Using `?.` here may potentially hide an issue, and the benchmark then wouldn't measure Objective-C interop.
Adding nullability annotations to Objective-C header may help to avoid the necessity to handle `null` here.